components/os/unix.go

```diff
@@ -48,34 +46,40 @@ func getDefaultInstanceType(env config.Environment, arch Architecture) string {
 	}
 }
 
-func getUnixInstallFormatString(scriptName string, version AgentVersion) string {
-	if version.PipelineID != "" {
-		testEnvVars := []string{}
-		testEnvVars = append(testEnvVars, "TESTING_APT_URL=apttesting.datad0g.com")
-		// apt testing repo
-		// TESTING_APT_REPO_VERSION="pipeline-xxxxx-a7 7"
-		testEnvVars = append(testEnvVars, fmt.Sprintf(`TESTING_APT_REPO_VERSION="%v-a7 7"`, version.PipelineID))
-		testEnvVars = append(testEnvVars, "TESTING_YUM_URL=yumtesting.datad0g.com")
-		// yum testing repo
-		// TESTING_YUM_VERSION_PATH="testing/pipeline-xxxxx-a7/7"
-		testEnvVars = append(testEnvVars, fmt.Sprintf("TESTING_YUM_VERSION_PATH=testing/%v-a7/7", version.PipelineID))
-		commandLine := strings.Join(testEnvVars, " ")
-
-		return fmt.Sprintf(
-			`DD_API_KEY=%%s %v bash -c "$(curl -L https://s3.amazonaws.com/dd-agent/scripts/%v)"`,
-			commandLine,
-			scriptName)
+func getUnixRepositoryParams(version AgentVersion) string {
+	envVars := []string{}
+	switch version.Repository {
+	case TrialRepository, TestingRepository:
+		aptChannel := version.Channel
```

Comment:
`Channel` _is_ not restricted to the values defined here - any string value is a valid channel, the three values defined here are only to avoid repetitions.

Reply:
if `Channel` can be any string then the type should be `string` to avoid confusion.